src/gtc/numpy/npir_codegen.py

```diff
@@ -14,56 +14,62 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-import functools
 import textwrap
 from dataclasses import dataclass, field
-from typing import Any, Collection, Dict, Tuple, Union
+from typing import Any, Collection, Optional, Set, Tuple, Union
 
+from eve import SymbolTableTrait
 from eve.codegen import FormatTemplate, JinjaTemplate, TemplatedGenerator
-from eve.visitors import NodeVisitor
-from gt4py.definitions import Extent
 from gtc import common
 from gtc.numpy import npir
 
 
 __all__ = ["NpirCodegen"]
 
 
-def op_delta_from_int(value: int) -> Tuple[str, str]:
-    operator = ""
-    delta = str(value)
-    if value == 0:
-        delta = ""
-    elif value < 0:
-        operator = " - "
-        delta = str(abs(value))
-    else:
-        operator = " + "
-    return operator, delta
+def _paren_wrap(s: str) -> str:
```

Comment:
Just a suggestion: it looks like this function is always used to wrap the expression coming from dumping some interator's content, so you could also add the logic for dumping the iterator here. Something like:

```python
def dump_sequence(sequence, *, separator=", ", start="(", end=")"):
    return f"{start}{separator.join(sequence)}{end}"
```

Reply:
Nice idea!